zapcore/field.go

```diff
@@ -209,7 +209,7 @@ func encodeStringer(key string, stringer interface{}, enc ObjectEncoder) (retErr
 	// Try to capture panics (from nil references or otherwise) when calling
 	// the String() method, similar to https://golang.org/src/fmt/print.go#L540
 	defer func() {
-		if err := recover(); err != nil {
+		if rerr := recover(); rerr != nil {
```

Comment:
can we undo this rename?

```suggestion
		if err := recover(); err != nil {
```